src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1001,6 +1001,12 @@ void RaftPart::processAppendLogResponses(const AppendLogResponses& resps,
     LOG_EVERY_N(WARNING, 100) << idStr_ << "Only " << numSucceeded
                               << " hosts succeeded, Need to try again";
     usleep(1000);
+    {
+      std::lock_guard<std::mutex> g(raftLock_);
+      // wal could be rollback between two cycles, if we still use the old lastLogId, we may always
+      // get E_RAFT_NO_WAL_FOUND
+      lastLogId = std::min(wal_->lastLogId(), lastLogId);
```

Comment:
There are two question here:
1. If you only modify lastLogId, but the `iter` is not modified, I really don't know whether we can succeed.
2. We will check the if the term has changed in `replicateLogs` (if rollback really happen, the term must have been updated), so I really doubt whether the problem should be fixed in this way. Or you could describe it in more details.

Reply:
we got E_RAFT_NO_WAL_FOUND  because there are some logs we are going to send don't exist in wal, the reason is that we alway use the old lastLogId, the lastLogId may be greater than part_->wal()->lastLogId().so update the lastLogId every circle, and if the wal is updated in this circle, continue in next circle with new lastLogId, and maybe there is a time that part_->wal() would not be updated and we send the logs.

Reply:
actually,  host use a new iterator from the wal, the iter is not used any more.